controller/client/controller_client.go

```diff
@@ -116,6 +116,33 @@ func (c *ControllerClient) CreateQuorumReplica(address string) (*rest.Replica, e
 	return &resp, err
 }
 
+func (c *ControllerClient) DeleteVolume() (*rest.DeleteReplicaOutput, error) {
+	var resp rest.DeleteReplicaOutput
+	vol, err := c.GetVolume()
+	if err != nil {
+		return nil, err
+	}
+	if len(vol.Actions["deletevolume"]) == 0 {
+		return nil, fmt.Errorf("No volume found")
+	}
+	httpReq, err := http.NewRequest("DELETE", vol.Actions["deletevolume"], nil)
+	if err != nil {
+		return nil, err
+	}
+	httpResp, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+	if httpResp.StatusCode >= 300 {
```

Comment:
Wait... >= 300 is a bad response? How so? What errors can we get back, to begin with? Also is this method synchronous? What if the volume is large? Would it not time out? Should we not implement this?

Reply:
As per the handling we will only get, 200, 404(volume not found), 409(Replication factor conflict) and 500(Bad response from replica), So we should handle this case to reply the client about the cause of error. 
Regarding the timeout I haven't tested it on the larger volumes. It uses SYS_UNLINKAT internally, i guess it mark the file as delete and comes out and deletion of the contents happens parallelly across replicas.

Reply:
So how is the user to understand, the differences between 404 and lets sat 409? 

As for deleting "large" files -- we would need to implement this async. So we post the request to delete it, get back a 202 (from memory, we should double check that) and use the resulting URL, to poll the status of the request.

Reply:
One of the main challenge is, if we start deleting the replica all the fields are set to nil, So replica IP will be removed from the controller. So if someone wants to see the status of the volume they can get at `REPLICA_IP:9502/v1/delete`
Following are the status:
"In Progress"
"Completed"
"(Error msg in case of any error)"

So it will reply the above msg's for the delete call for the first request based on the status of replica at that point of time.

Reply:
I'm not sure I understand what you are trying to say, sorry my bad. Not sure if you understand my comment either, but to hopefully make less unclear. 

If we are accepting a delete request, which we process in the background, the server -- should have a queue of work in progress that the client can poll. This is a pretty common approach in rest land. The response object of the request contains an URL, which the callee can poll until it completes (or fails).

I hope this makes sense to you

Reply:
@gila i going with async approach where it will start deleting the contents of the volume in the background and it will return "InProgress" immediately. So even if someone makes a delete call again, it will show "resourse is busy" (if deletion is still happening in the background) else it will return error saying "s.r = nil".
We will implement at the delete operation via polling later in some other PR.

Reply:
That's sounds OK 👍